solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -233,6 +233,41 @@ public void prepare(ResponseBuilder rb) throws IOException
     }
   }
 
+  private boolean allowSkipSecondGroupingStep(final GroupingSpecification groupingSpec, final boolean isReranking ) {
+    // Only possible if we only want one doc per group
+    if (groupingSpec.getGroupLimit() != 1) {
+        logger.error("group.skip.second.step=true is not compatible with group.limit == " + groupingSpec.getGroupLimit() );
+        return false;
+    }
+
+    // Within group sort must be the same as group sort because if we skip second step no sorting within group will be done.
+    if (groupingSpec.getSortWithinGroup() !=  groupingSpec.getGroupSort()) {
+        logger.error("group.skip.second.step=true is not compatible with group.sort != sort");
+        return false;
+    }
+
+    boolean byRelevanceOnly = false;
+    SortField[] sortFields = groupingSpec.getGroupSort().getSort();
+
+    if(sortFields != null && sortFields.length == 1 && sortFields[0] != null && sortFields[0].getComparator() instanceof FieldComparator.RelevanceComparator) {
```

Comment:
SortField => getComparator(final int numHits, final int sortPos) .. differs in signature of the api..

Reply:
Yeah, the PR went out of sync, we will update it asap

Reply:
> Yeah, the PR went out of sync ...

If you'd like https://github.com/bloomberg/lucene-solr/pull/228 pull request to merge into the https://github.com/bloomberg/lucene-solr/tree/SOLR-11831 branch (the branch behind this pull request here) has an interim 'make it compile (somehow)' change (also includes 'logger' to 'LOG' renames in this method).